### PR TITLE
Various improvements and cleanup

### PR DESCRIPTION
**Pull request by amerkoleci**
Geneate xml documentation only when compling in Release.
Update Microsoft.NETCore.UniversalWindowsPlatform to 5.4.0.
Remove not needed netcoreapp1.0 target framework.
Cleanup some macro to use WINDOWS_UWP instead of STORE_APP.
XAudio fix desktop Version 2.7 on non UWP platform.
Direct3D12 some documentation improvements and some helper logic.

### CODE REVIEW

**Comment by amerkoleci** [comment]
This fixes also https://github.com/MonoGame/MonoGame/issues/5986

File: Source/SharpDX/Mapping.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
   <depends>sharpdx-core</depends>
   <depends>sharpdx-core-direct3d</depends>
   <depends>sharpdx-core-multimedia</depends>
-
+l
```
Comment (xoofx) [inline Source/SharpDX/Mapping.xml]: There is a 1 here?

Comment (amerkoleci) [reply Source/SharpDX/Mapping.xml]: My bad, fixed

**Comment by xoofx** [comment]
The build is failing, could you have a look?

**Comment by amerkoleci** [comment]
Strange, seams tha appveyor cannot resolve uap10.0 target, can there be any settings?

I updated uwp platform to 10.0.14393.0

**Comment by amerkoleci** [comment]
Locally I can build without issues

**Comment by amerkoleci** [comment]
Fixed now

**Comment by amerkoleci** [comment]
Fixes also the native effects dlls path, see #941

**Comment by xoofx** [comment]
I would prefer that you remove 7dcca8b as part of this PR. The reason is that I didn't want Direct3D12 to follow the previous legacy API (that inherited from the SlimDX decisions) and wanted them to be much closer to the underlying C++ code. That's why there is little manual code written for Direct3D12 and I would like to keep it like this.

In the future, If you could split the PR per issue addressed, it would help to let them go through more easily.

**Comment by amerkoleci** [comment]
I will revert the commit, i know but as i coded I saw the issue and decided to fix it :) but will submit separated PR in future

**Comment by amerkoleci** [comment]
Should be good now

**Comment by xoofx** [comment]
Thanks!

**Pull request merged**